src/lib.rs

```diff
@@ -70,8 +70,16 @@
 
 #![deny(warnings, missing_docs, missing_debug_implementations)]
 #![doc(html_root_url = "https://docs.rs/bytes/0.4")]
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(feature = "nightly", feature(alloc))]
```

Comment:
@alexcrichton so really this is the part that requires nightly, and that's because even if I have an `#[allocator]` defined `Box`, `Vec`, and `String` do not get added to the prelude.

If they did, then we could simply have an enabled-by-default, non-nightly-dependent `alloc` cargo feature  for gating features that depend on `Box`, `Vec`, and `String`.

I'm not sure how hard that would be, but I imagine it would be immensely useful for `no_std` users who have a heap available.